E2E tests read bedspace attributes from page

bryangaledxw opened the pull request:
# Changes in this PR

- We update to UI E2E tests, so we no longer need to keep `characteristics.json` file synchronised with the API in order for the tests to pass. We instead derive this data from the bedspace creation and editing pages

# Release checklist

[Release process
documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually
  approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

<!-- Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible. You can monitor deployment failures in CircleCI
itself and application errors are found in
[Sentry](https://ministryofjustice.sentry.io/issues/?project=4504129156218880&referrer=sidebar&statsPeriod=24h).
Both events should be automatically sent to our [Slack
channel](https://mojdt.slack.com/archives/C048BJS7S2F). -->

bryangaledxw committed:
Sort imports

bryangaledxw committed:
Add assignCharacteristics to BedspaceEditable test helper

This will enable us to derive characteristic data from the new bedspace and update bedspace pages

bryangaledxw committed:
Update room related factories

We add helper methods in our room related factories to reduce duplication in our E2E tests

bryangaledxw committed:
Update bedspace E2E tests

We update our bedspace E2E tests to use our new factory helper methods, and to read characteristic data from the page, rather than relying on a characteristics.json file that's kept in sync with the API

patrickjfl commented on cypress_shared/pages/temporary-accommodation/manage/bedspaceEditable.ts:
```diff
@@ -12,4 +12,8 @@ export default abstract class BedspaceEditablePage extends Page {
     this.clickSubmit()
   }
+
+  getCharacteristics(alias: string): void {
```
If this method is only ever likely to assign something to the alias 'characteristics', do we need the alias parameter? I wonder if it's more confusing than helpful. An alternative might be to call the method assignCharacteristicsToAlias and then pass 'characteristics' in to the method called inside.

bryangaledxw replied:
I think I'd rather pass in the alias - keep the alias name knowledge in one place (in the step definitions, rather than both the step definitions and the page helper class). But I think changing "get" to "assign" is a good call. How would you feel about just `assignCharacteristics(alias: string)` and the "ToAlias" is implicit by context/convention?

patrickjfl replied:
yep sure, that makes sense :)

patrickjfl reviewed (approved):
One suggestion but LGTM 👍

bryangaledxw committed:
Rename E2E test page reading methods

We have a number of helper methods for reading data from a page in our E2E tests, then assigning that data to an alias. We rename these methods for clarity
